src/realm/error_codes.cpp

```diff
@@ -60,11 +68,56 @@ StringData ErrorCodes::error_string(Error code)
         case ErrorCodes::WebSocket_TLSHandshakeFailed:
             return "WebSocket: TLS Handshake Failed";
 
+        /// WebSocket Errors - reported by server
+        case ErrorCodes::WebSocket_Unauthorized:
+            return "WebSocket: Unauthorized";
+        case ErrorCodes::WebSocket_Forbidden:
+            return "WebSocket: Forbidden";
+        case ErrorCodes::WebSocket_MovedPermanently:
+            return "WebSocket: Moved Permanently";
+        case ErrorCodes::WebSocket_Client_Too_Old:
+            return "WebSocket: Client Too Old";
+        case ErrorCodes::WebSocket_Client_Too_New:
+            return "WebSocket: Client Too New";
+        case ErrorCodes::WebSocket_Protocol_Mismatch:
+            return "WebSocket: Protocol Mismatch";
+
         case ErrorCodes::UnknownError:
             [[fallthrough]];
         default:
             return "UnknownError";
     }
 }
 
+namespace {
+
+class CloseStatusErrorCategory : public std::error_category {
```

Comment:
Since these are generic Status code values, make this `StatusErrorCategory` and then rename the function to `status_error_category()` instead of `websocket_close_status_category()`